### ISSUE TITLE
[BUG] Can't change the network for an existing VM, even when powered down

### ISSUE THREAD

**Issue by egrigson** [harvester/harvester#1914]
**Describe the bug**
On an existing VM you can't change the assigned network - for example from 'management' to 'VLAN' (masquerade to bridge)

**To Reproduce**
1. Create a VM with at least one network interface
2.Go to Virtual Machines > Edit Config > Networks and you'll see that the only editable field is 'Type' - but NOT 'Network' 

**Expected behavior**
I'd expect to be able to use the 'Network' dropdown to select any of my predefined networks (and ideally create a new network if desired for bonus points). This should work whether the VM is powered on or off.

**Environment:**
 - Harvester ISO version: 1.0.0
 - Underlying Infrastructure (e.g. Baremetal with Dell PowerEdge R630): nested in vSphere v7.0

**Additional context**
None

**Comment by FrankYang0529** [comment]
~~Harvester works fine for changing network from vlan to management, but there is error from management to vlan.~~

~~Reproduce steps:~~
~~1. Create a VM with management network.
2. When VM is running, click `Edit config`.
3. Add a VLAN network and remove management network. Click `Save` to restart VM.
4. Although VM can run, we don't have an IP address for the VM.~~

**Comment by simonflood** [comment]
It doesn't matter whether VM is running or shutdown, you can't change from management to VLAN (perhaps after you've configured the VLAN in Harvester). However after downloading and comparing the YAML files for a VM using management VLAN and another using VLAN, it's possible to do this by editing the YAML:

1. change `- masquerade: {}` under `interfaces:` to `- bridge: {}`
2. change the `networks:` section to

<pre>
networks:
- multus:
    networkName: default/name_of_your_network
  name: default
</pre>

**Comment by FrankYang0529** [comment]
@simonflood , you are right. We can change the network by YAML, but we still need to restart the VM to apply the change. Please see the first video.

**Case 1: change network by YAML**
In this case, if we restart the VM after we modify VM YAML, the pod will not be updated. After restarting the VM, we can get VLAN network with an IP.

https://user-images.githubusercontent.com/4927361/154406863-358ed6b3-f7da-4d3a-915d-75ad538e3f4e.mov

**Case 2: change network by UI**
We have a bug here. Not sure why we can't get an IP for VM after updating the network. cc @guangbochen 
1. Create a VM with a management network.
2. After it's running, click `Edit Config`.
3. In `Network` tab, remove the management network, add VLAN network and change its name to default.
4. Click Save.
5. After restarting, VM can't get an IP.

https://user-images.githubusercontent.com/4927361/154407345-3a3915be-d727-4dda-a378-0f2526a914c9.mov

**Comment by w13915984028** [comment]
The `Case 2` needs a further investigation. @FrankYang0529 @yaocw2020 @guangbochen 

```
Case 2: change network by UI
We have a bug here. Not sure why we can't get an IP for VM after updating the network. cc @guangbochen

Create a VM with a management network.
After it's running, click Edit Config.
In Network tab, remove the management network, add VLAN network and change its name to default.
Click Save.
After restarting, VM can't get an IP.
```

**Comment by guangbochen** [comment]
For case2, the VLAN IP can only be retrieved when the `qemu-agent` is installed, but the `qemu-agent` will only be installed on VM's init boot, that could be the potential issue. But users can reinstall it per https://docs.harvesterhci.io/v1.0/faq/#how-to-install-the-qemu-guest-agent-of-a-running-vm

**Comment by FrankYang0529** [comment]
Hi @guangbochen @w13915984028, I test again through UI, but this time I add Mac Address from the management Network to a VLAN network. After restarting the VM, it can get an IP.

![Screen Shot 2022-07-19 at 9 04 45 AM](https://user-images.githubusercontent.com/4927361/179642020-5092e994-e26e-41f9-90e2-2d3e25fc9cd4.png)

**Comment by w13915984028** [comment]
@FrankYang0529 thanks, it meets our analysis.

I would suggest we enhance it via:
(1) add `modify` network ability to the UI, user can modify instead of `remove+add`.  (if this is possible)
(2) add note and guide to `Harvester Doc`, that `change` VM network needs special care, to keep the `MAC` unchanged.

@guangbochen

**Comment by rrajendran17** [comment]
@w13915984028 @FrankYang0529 
"(2) add note and guide to Harvester Doc, that change VM network needs special care, to keep the MAC unchanged."
Is this still valid on latest releases ?can you please detail on why a mac should be unchanged when we change the interface type for a vm ?

1.I tried creating a vm and added mgmt interface masquerade mode on v1.3.1
2. once vm is up and running and I edited the config and changed to bridge type specifying a vlan vm network 
3.  save and restarted the vm
4. vm able to get ip address.
I did not configure any mac from UI and it was able to get the ip correctly.

I am asking this as I am debugging a similar customer issue where its a windows vm and after changing from mgmt to bridge interface on a vm, the interface is not able to form any network connectivity.
The reason I see in this case is ip is not acquired by this interface, as the mac is not copied from the vm pod interface to the windows ethernet interface and tap interface is dropping packets.
Reference : https://jira.suse.com/browse/SURE-9017

**Comment by rrajendran17** [comment]
@a110605 @torchiaf do you think we need to fix the UI issue ?
  Currently we cannot edit and modify network config one network to another, but we could add a new nic and remove the old one.

@egrigson I do not see the ip address issue after changing from masquerade(mgmt) to bridge(vlan network) in the latest.Can you check on latest ?

**Comment by torchiaf** [comment]
> @a110605 @torchiaf do you think we need to fix the UI issue ? Currently we cannot edit and modify network config one network to another, but we could add a new nic and remove the old one.
> 
I'm in favor of adding the ability to the UI to modify the networks as long as this is possible via YAML edit. We should also put a warning banner or validation steps when edit VM Networks.
